compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -365,12 +365,10 @@ fn check_region_bounds_on_impl_item<'tcx>(
         let item_kind = assoc_item_kind_str(impl_m);
         let def_span = tcx.sess.source_map().guess_head_span(span);
         let span = tcx.hir().get_generics(impl_m.def_id).map_or(def_span, |g| g.span);
-        let generics_span = if let Some(sp) = tcx.hir().span_if_local(trait_m.def_id) {
+        let generics_span = tcx.hir().span_if_local(trait_m.def_id).map(|sp| {
             let def_sp = tcx.sess.source_map().guess_head_span(sp);
-            Some(tcx.hir().get_generics(trait_m.def_id).map_or(def_sp, |g| g.span))
-        } else {
-            None
-        };
+            tcx.hir().get_generics(trait_m.def_id).map_or(def_sp, |g| g.span)
```

Comment:
technically we only want to compute the `guess_head_span` if `get_generics` is `None`, so we could use `map_or_else` here (or just open code a match, which may be more readable.

Reply:
I noticed that but we're already in the error case so I left it alone. Can still change it though, `map_or_else` wasn't particularly bad IIRC.

Reply:
Ah right, yea just leave it as it is then.